### PR TITLE
Enhance HTML report with support for other models and timeline chart

### LINKED ISSUE #42: Wrong Date and Time in the clipboard output
screenshot_2021-07-11_19-51

### PR DESCRIPTION
**Pull request by ljmf00**
* expression history
* clipboard
* translation history

Also refactored diet views for more comprehensive and readable
templating.

Signed-off-by: Luís Ferreira <contact@lsferreira.net>

---
Fixes #42 .

### CODE REVIEW

**Comment by ljmf00** [comment]
:angry: Why heisenberg's...

**Comment by iK4tsu** [comment]
force push to restart the CI

File: source/gboardforensics/gatherers/clipboard.d [hunk 1]
```diff
@@ -33,16 +38,35 @@ class ClipboardGatherer : IGatherer
...
+
+		db.execute(`SELECT
 				text,
 				html_text AS html,
 				uri IS NULL OR uri = "" AS type,
 				datetime(timestamp/1000, 'unixepoch') AS time,
 				timestamp,
-				uri
+				uri,
+				""
```
Comment (iK4tsu) [inline source/gboardforensics/gatherers/clipboard.d]:
```suggestion
				"" AS content
```

Comment (iK4tsu) [reply source/gboardforensics/gatherers/clipboard.d]: This is way easier to understand what is the purpose of the field. I would advise you to also document it.

File: source/gboardforensics/gatherers/clipboard.d [hunk 2]
```diff
@@ -33,16 +38,35 @@ class ClipboardGatherer : IGatherer
...
-			.array;
+			.each!((Clipboard.Entry e) {
+				if(e.type == Clipboard.Entry.Type.DOCUMENT)
+				{
+					auto documentPath = buildPath(
+						dirName(_clipboard.path),
+						"../files/" ~ e.uri
+							.findSplitAfter("content://")[1]
+							.findSplitAfter("/")[1]
+					);
```
Comment (iK4tsu) [inline source/gboardforensics/gatherers/clipboard.d]:
```suggestion
					auto documentPath = _clipboard.path
						.dirName
						.buildPath("..", "files", e.uri
							.findSplitAfter("content://")[1]
							.findSplitAfter("/")[1]
					);
```

**Review by iK4tsu — CHANGES_REQUESTED** [review]

File: source/gboardforensics/gatherers/clipboard.d (hunk 1 above)
Comment (ljmf00) [reply source/gboardforensics/gatherers/clipboard.d]: Done

File: source/gboardforensics/gatherers/clipboard.d (hunk 2 above)
Comment (ljmf00) [reply source/gboardforensics/gatherers/clipboard.d]: Done

**Review by iK4tsu — APPROVED** [review]

**Pull request merged**